File: src/main/java/com/zerozero/store/infrastructure/mongodb/StoreMongoRepository.java
package com.zerozero.store.infrastructure.mongodb;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import java.util.List;
import java.util.UUID;

public interface StoreMongoRepository extends MongoRepository<Store, UUID> {

    @Query("{ 'location': { $near: { $geometry: { type: 'Point', coordinates: [?0, ?1] }, $maxDistance: ?2 } } }")
    List<Store> findStoresWithinCoordinatesRadius(double longitude, double latitude, double maxDistance);
}

File: src/main/java/com/zerozero/store/infrastructure/mongodb/StoreMongoRepository.kt
package com.zerozero.store.infrastructure.mongodb

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import java.util.*

interface StoreMongoRepository : MongoRepository<Store, UUID> {

    @Query("{ 'location': { \$near: { \$geometry: { type: 'Point', coordinates: [?0, ?1] }, \$maxDistance: ?2 } } }")
    fun findStoresWithinCoordinatesRadius(longitude: Double, latitude: Double, maxDistance: Double): List<Store>
}
